src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/items/SoulboundRune.java

```diff
@@ -0,0 +1,118 @@
+package me.mrCookieSlime.Slimefun.Objects.SlimefunItem.items;
+
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.Sound;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Item;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Lists.SlimefunItems;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.Objects.Research;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SimpleSlimefunItem;
+import me.mrCookieSlime.Slimefun.Objects.handlers.ItemDropHandler;
+import me.mrCookieSlime.Slimefun.Setup.Messages;
+import me.mrCookieSlime.Slimefun.Setup.SlimefunManager;
+import me.mrCookieSlime.Slimefun.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.api.PlayerProfile;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class SoulboundRune extends SimpleSlimefunItem<ItemDropHandler> {
+
+    public SoulboundRune(Category category, ItemStack item, String id, RecipeType type, ItemStack[] recipe) {
+        super(category, item, id, type, recipe);
+    }
+
+    @Override
+    public ItemDropHandler getItemHandler() {
+        return (e, p, i) -> {
+            ItemStack item = i.getItemStack();
+            // We are using a boolean array because we will change the boolean's value inside a lambda
+            // but you can't access non-final variables from outside the lambda inside the lambda.
+            final boolean[] boo = {false};
```

Comment:
Yes I should and I *must*

Reply:
No, you shouldn't, you must not and you cannot.

Again, you are always returning false, because of your async invokation.
This boolean array right there does nothing, it is pointless.

Reply:
Although it's not async. all the invokations in there are sync

Reply:
" Bukkit.getScheduler().scheduleSyncDelayedTask "
is an asynchronous call.

To illustrate:
You are passing the Runnable to be run 20 ticks after this method finished.
This is what is called an "asynchronous call".
The return-statement will have already fired long before the code in your lambda expression will even be run.
You are not halting the entire server for 20 ticks, that'd be ridiculous.

So:
The return-statement was fired, then you are modifying an array that does absolutely nothing, because it's value has already been returned.

This is also the reason why "it doesn't let you use a normal boolean".



Still don't believe me?
Print the output of this method to the console, I guarantee you it's gonna be always false.